ocaml/xapi/message_forwarding.ml

```diff
@@ -6586,7 +6586,7 @@ functor
         let fn ~rpc:_ ~session_id:_ ~host =
           do_op_on ~__context ~host ~local_fn client_fn
         in
-        Xapi_pool_helpers.call_fn_on_slaves_then_master ~__context fn
+        Xapi_pool_helpers.call_fn_on_master_then_slaves ~__context fn
```

Comment:
why are we reversing the order?

Reply:
I explained in the commit message: there was no specific reason for doing slaves then master and reversing it means that master (and therefore the DB) gets updated before all the forwarders. This means the Dom0ObserverConfig can rely on the data in the DB to update the config files. If we don't reverse it, the DB is updated last so the config files are based on old info.